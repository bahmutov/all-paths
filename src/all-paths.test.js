'use strict'

/* eslint-env jest */
const allPaths = require('.')

test('single property', () => {
  const o = {
    foo: 'bar'
  }
  expect(allPaths(o)).toEqual([['foo']])
})

test('plain object', () => {
  const o = {
    foo: { bar: 42 }
  }
  expect(allPaths(o)).toMatchSnapshot()
})

test('complex object', () => {
  const o = {
    foo: { bar: 42 },
    list: ['one', 'two', 'three']
  }
  expect(allPaths(o)).toMatchSnapshot()
})
