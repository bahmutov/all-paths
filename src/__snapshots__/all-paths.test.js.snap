// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`complex object 1`] = `
Array [
  Array [
    "foo",
  ],
  Array [
    "foo",
    "bar",
  ],
  Array [
    "list",
  ],
  Array [
    "list",
    0,
  ],
  Array [
    "list",
    1,
  ],
  Array [
    "list",
    2,
  ],
]
`;

exports[`plain object 1`] = `
Array [
  Array [
    "foo",
  ],
  Array [
    "foo",
    "bar",
  ],
]
`;
